Lib/_pyio.py

```diff
@@ -1577,7 +1577,11 @@ def __init__(self, file, mode='r', closefd=True, opener=None):
                 # For consistent behaviour, we explicitly seek to the
                 # end of file (otherwise, it might be done only on the
                 # first write()).
-                os.lseek(fd, 0, SEEK_END)
+                try:
+                    os.lseek(fd, 0, SEEK_END)
+                except OSError as e:
+                    if e.errno != errno.ESPIPE:
```

Comment:
Is errno.ESPIPE available on common platforms (Windows, Linux, macOS, FreeBSD)?

Reply:
It's certainly available on Unixes and the Windows tests are passing.

Reply:
Great!